bokehjs/src/lib/models/widgets/file_input.ts

```diff
@@ -103,12 +126,14 @@ export class FileInput extends InputWidget {
   static {
     this.prototype.default_view = FileInputView
 
-    this.define<FileInput.Props>(({Bool, Str, List, Or}) => ({
+    this.define<FileInput.Props>(({Bool, Str, List, Or, Int}) => ({
       value:     [ Or(Str, List(Str)), p.unset, {readonly: true} ],
       mime_type: [ Or(Str, List(Str)), p.unset, {readonly: true} ],
       filename:  [ Or(Str, List(Str)), p.unset, {readonly: true} ],
       accept:    [ Or(Str, List(Str)), "" ],
       multiple:  [ Bool, false ],
+      directory: [ Bool, false ],
+      _clear_input: [ Int, 0],
```

Comment:
This is not the final design (I hope). 

I need help figuring out how to tell the input to be cleared from Python to BokehJS. Can an `event` be used?

One of the problems, at least for me, is that `value` is read-only from the Python side.